# origin: chrisw (BitKeeper)
# cset: 1.1484 (2.4) key=4244a3d91w2q8hQzZefX9zsDlTlJtA
# URL: http://linux.bkbits.net:8080/linux-2.4/cset@4244a3d91w2q8hQzZefX9zsDlTlJtA
# inclusion: upstream
# descrition: [PATCH] isofs: more "corrupted iso image" error cases
# revision date: Tue, 29 Mar 2005 17:41:06 +0900
#
# S rset: ChangeSet|1.1483..1.1484
# I rset: fs/isofs/inode.c|1.10..1.11
#
# Key:
# S: Skipped  ChangeSet file only
# O: Original Followed by Updated
# U: Updated  Included with updated range of versions
# I: Included Included verbatim
# E: Excluded Excluded on request from user
# D: Deleted  Manually deleted by subsequent user edit
# R: Revised  Manually revised by subsequent user edit
#
#
# This is a BitKeeper generated diff -Nru style patch.
#
# fs/isofs/inode.c
#   2005/03/26 00:41:03-03:00 chrisw@osdl.org +2 -0
#   isofs: more "corrupted iso image" error cases
# 
# ChangeSet
#   2005/03/25 20:50:49-03:00 chrisw@osdl.org 
#   [PATCH] isofs: more "corrupted iso image" error cases
#   
#   Michal Zalewski <lcamtuf@dione.ids.pl> discovers range checking flaws in
#   iso9660 filesystem.
#   
#   http://marc.theaimsgroup.com/?l=bugtraq&m=111110067304783&w=2
#   
#   CAN-2005-0815 is assigned to this issue.
#   
#   From: Linus Torvalds <torvalds@osdl.org>
#   
#   isofs: more "corrupted iso image" error cases
#   
#   Thanks to Michal Zalewski for testing.
#   
#   Signed-off-by: Chris Wright <chrisw@osdl.org>
# 
#
===== fs/isofs/inode.c 1.10 vs 1.11 =====
--- 1.10/fs/isofs/inode.c	2004-11-19 03:11:56 +09:00
+++ 1.11/fs/isofs/inode.c	2005-03-26 12:41:03 +09:00
@@ -609,6 +609,8 @@ root_found:
 	  s->u.isofs_sb.s_log_zone_size = isonum_723 (h_pri->logical_block_size);
 	  s->u.isofs_sb.s_max_size = isonum_733(h_pri->volume_space_size);
 	} else {
+	  if (!pri)
+	    goto out_freebh;
 	  rootp = (struct iso_directory_record *) pri->root_directory_record;
 	  s->u.isofs_sb.s_nzones = isonum_733 (pri->volume_space_size);
 	  s->u.isofs_sb.s_log_zone_size = isonum_723 (pri->logical_block_size);
